src/sotoki/archives.py

```diff
@@ -89,7 +90,12 @@ def _run(url, fpath):
 
             # remove other files from ark that we won't need
             for fp in self.build_dir.iterdir():
-                if fp.suffix == ".xml" and fp.stem not in self.dump_parts:
+                if fp.suffix == ".xml":
+                    if fp.stem not in self.dump_parts:
+                        fp.unlink()
+                    else:
+                        reencode_file(fp)
```

Comment:
This is in a block declared as a cleanup step while this is mandatory preparatory work.